src/inputs/SelectField.test.tsx

```diff
@@ -154,17 +149,74 @@ describe("SelectFieldTest", () => {
     expect(onSelect).not.toHaveBeenCalled();
   });
 
+  it("can load options via options prop callback", async () => {
+    // Given a Select Field with options that are loaded via a callback
+    const r = await render(
+      <TestSelectField
+        label="Age"
+        value="1"
+        options={{ initial: [options[0]], load: async () => ({ options }) }}
+        getOptionLabel={(o) => o.name}
+        getOptionValue={(o) => o.id}
+        data-testid="age"
+      />,
+    );
+    // When opening the menu
+    fireEvent.focus(r.age());
+    // Then expect to see the initial option and loading state
+    expect(r.getAllByRole("option")).toHaveLength(1);
+    expect(r.loadingDots()).toBeTruthy();
+    // And when waiting for the promise to resolve
+    await wait();
+    // Then expect the rest of the options to be loaded in and the loading state to be removed
+    expect(r.getAllByRole("option")).toHaveLength(3);
+    expect(r.queryByTestId("loadingDots")).toBeFalsy();
+  });
+
+  it("reflects new options when prop changes", async () => {
+    // Given a Select Field with options that are loaded via a callback
+    const r = await render(
+      <TestSelectField
+        label="Age"
+        value="1"
+        options={[options[0]]}
+        getOptionLabel={(o) => o.name}
+        getOptionValue={(o) => o.id}
+        data-testid="age"
+      />,
+    );
+    // When opening the menu
+    fireEvent.focus(r.age());
+    // Then expect to see the initial option
+    expect(r.getAllByRole("option")).toHaveLength(1);
+    // And when changing the options
+    click(r.updateOptions);
```

Comment:
Ouu how does this work?

Reply:
Not 100% what you're referring to when you ask this. Basically, I put a button in the `TestSelectField` component defined in the test that would change the `options` prop value.